core_contracts/governance/governance.py

```diff
@@ -28,30 +28,108 @@ class Governance(IconScoreBase):
     def __init__(self, db: IconScoreDatabase) -> None:
         super().__init__(db)
         self.addresses = Addresses(db, self)
+        self.vote_execute = VoteActions(db, self)
         self._launch_day = VarDB('launch_day', db, int)
         self._launch_time = VarDB('launch_time', db, int)
         self._launched = VarDB('launched', db, bool)
+        self._votes = DictDB('votes', db, int, depth=3)
+        self._total_voted = DictDB('total_voted', db, int, depth=2)
+        self._minimum_vote_duration = VarDB('min_duration', db, int)
+        self._vote_actions = DictDB('vote_actions', db, str)
+        self._vote_count = VarDB('vote_count', db, int)
 
     def on_install(self) -> None:
         super().on_install()
         self._launched.set(False)
 
     def on_update(self) -> None:
         super().on_update()
+        self._minimum_vote_duration.set(42300)
 
     @external(readonly=True)
     def name(self) -> str:
         return "Balanced Governance"
 
+    @external
+    @only_owner
+    def defineVote(self, name: str, quorum: int, duration: int, actions: str, majority: int = MAJORITY) -> None:
+        """
+        Names a new vote, defines quorum, and actions.
+        """
+        if 0 > quorum > 100:
+            revert(f'Quorum must be greater than 0 and less than 100.')
+        min_duration = self._minimum_vote_duration.get()
+        if duration < min_duration:
+            revert(f'Votes must have a minimum duration of {min_duration} blocks.')
+        if name in self._votes[0]:
+            revert(f'Poll name {name} has already been used.')
+        vote_index = self._vote_count.get() + 1
+        self._vote_count.set(vote_index)
+        self._votes[0][name]['id'] = vote_index
+        self._votes[vote_index][0]['active'] = 1
+        self._votes[vote_index][0]['quorum'] = quorum
+        self._votes[vote_index][0]['majority'] = majority
+        self._votes[vote_index][0]['last_block'] = self.block.height + duration
+        self._vote_actions[vote_index] = actions
+
+    @external
+    def castVote(self, name: str, vote: bool) -> None:
+        """
+        Casts a vote in the named poll.
+        """
+        if name not in self._votes[0] or not self._votes[self._votes[0][name]['id']][0]['active']:
```

Comment:
self._votes is a dictdb. Can't perform `in` check here.

Reply:
DictDB has a __contains__ method, so `in` works.